### DISCUSSION TITLE
Navy Rework

### DISCUSSION THREAD

**Discussion by OcelotWalrus** [OpenHV/OpenHV#954] [Ideas]
I've gotten some ideas when trying to design a only-water map, that could, in my opinion, really enhance the naval gameplay as for now, it's pretty limited: you could do a pod/vehicle-only map, and it'd work perfectly, you could do a aircraft-only map, it'd work perfectly; but you couldn't do a naval-only map.

The goal of this discussion is to discuss and find out what can be done to enhance the naval gameplay, before I start creating a PR, in order so that they actually have as enough importance as aircraft and pod/vehicles in the army. Because, for now, building an harbor really doesn't have any interest: vehicles are great because they're pretty cheap and they can be built in series fast, aircraft are useful because they can defend pretty much every far point of the map, pretty fast, but ships, they don't really have a point, unless maybe the missile submarine and the drone ship.

The ideas I have are the following ones:

* [x] A resource that can be placed on water tiles (I was thinking about oil), and that a boat similar to the miner could go on and be deployed, as a mining tower.
* [x] A new refinery that can go onto water (not much idea about this one)
* [x] A new boat that would work like the radar vehicle
* [x] Add more ship variants, to kinda fit with the existing pod/vehicle variants
* [ ] Some sort of a setup to allow navy units pass under bridges. This one cripples naval map design on it's own alone
*  [ ] Add a water builder?

_What I've explained here may not be really clear, so just comment if there's something you can't figure out._

**Comment by GraionDilach** [comment]
Wow, a lot of ideas and neither mention the most limiting fact wrt navy:

Some sort of a setup to allow navy units pass under bridges. This one cripples naval map design on it's own alone.

The thing is that going towards Warcraft 2 styled naval options (with oil and whatnot) won't help on the depth alone. While Warcraft 2 navy had more depth than even RA2 due to the dedicated resource on paper, navy in War2 became more of a gimmick than in RA1 or RA2. Red Alert 3's building spam could work but it needs a lot of amphibious options in order to ensure it would be designwise balanced - not gameplay balanced, just designwise balanced into something where the player doesn't feel it as a pointless gimmick yet doesn't feel necessary enough to refuse giving up on due to say, the map lacking water for the build tree either.

**Comment by OcelotWalrus** [comment]
> Some sort of a setup to allow navy units pass under bridges. This one cripples naval map design on it's own alone.

Yes I didn't really thought about that. The bridge tiles would need to be converted in some kind of actor then.

> The thing is that going towards Warcraft 2 styled naval options (with oil and whatnot) won't help on the depth alone. While Warcraft 2 navy had more depth than even RA2 due to the dedicated resource on paper, navy in War2 became more of a gimmick than in RA1 or RA2. Red Alert 3's building spam could work but it needs a lot of amphibious options in order to ensure it would be designwise balanced - not gameplay balanced, just designwise balanced into something where the player doesn't feel it as a pointless gimmick yet doesn't feel necessary enough to refuse giving up on due to say, the map lacking water for the build tree either.

Yes that's true. My point here is not to add some cool features to the navy, but make so that the navy gameplay gets as much role as the terrain and aircraft gameplays. What I mean by that it that the navy lacks in utility.

**Comment by GraionDilach** [comment]
Not necessarily, copying over the TS tunnels setup is also an option.

**Reply by OcelotWalrus** [reply]
TS tunnels? What are they?

**Reply by Mailaender** [reply]
They have portals on both sides and then leave the map grid while underneath.

**Reply by OcelotWalrus** [reply]
Ooh, I see.

**Comment by WilliamWeberBerrutti** [comment]
When "shallow" terrain is added (thinking in Age of Empires terms), both land and water units could cross it. This'll give a big potential for navy units.

Also we need maps with more water proportion I think so water play is encouraged.

> A new boat that would work like the radar vehicle

The new boat that works like a Radar should be interesting to be added!

> Why submarines de-cloak when attacking?

For balance reasons. Imagine a situation if submarines did not de-cloak while attacking: how the other player would react to it without detectors? The other player would require a higher tech, consequently more investment, to counter submarines (thinking about the basic one from Synapol), which will make things imbalanced.

> Add more ship variants, to kinda fit with the existing pod/vehicle variants

I like this idea too, but first things first: we have to properly test navy the way it is now, with new maps and water features before adding new things so it's less stressful to work on it later. At least the Radar Boat should be interesting to be added, and I think this one could be worked on before adding any other navy unit.

> Some sort of a setup to allow navy units pass under bridges. This one cripples naval map design on it's own alone

This one is interesting too, but maybe we should focus on shallow terrain first and see what happens.

**Reply by OcelotWalrus** [reply]
> When "shallow" terrain is added (thinking in Age of Empires terms), both land and water units could cross it. This'll give a big potential for navy units.

It'd be amazing.

> Also we need maps with more water proportion I think so water play is encouraged.

I agree, but many new water buildings/units/deposits have to be added in order to make maps where water combat is master.

**Reply by WilliamWeberBerrutti** [reply]
I just edited the comment haha. Read it again please.

**Reply by WilliamWeberBerrutti** [reply]
> I agree, but many new water buildings/units/deposits have to be added in order to make maps where water combat is master.

Not necessarily. If there's more water proportion, you could do only water units (or most of your army be composed of water units) and win the game. I playtested it a lot and navy now is powerful, they just need opportunities during gameplay

**Comment by OcelotWalrus** [comment]
> For balance reasons. Imagine a situation if submarines did not de-cloak while attacking: how the other player would react to it without detectors? The other player would require a higher tech, consequently more investment, to counter submarines (thinking about the basic one from Synapol), which will make things imbalanced.

That's were the radar ship could come in help. I should make a PR to add that radar ship pretty soon.

**Reply by GraionDilach** [reply]
A radar ship doesn't help upon a deadlock of the defending player already lacks/lost a shipyard and the subs are just turtling around. Subs decloaked on firing ensures that such a defending player's idle units with range will respond to the attack without the player necessarily resorting to force-attack blindly.

**Reply by OcelotWalrus** [reply]
You do make a good point.

**Comment by OcelotWalrus** [comment]
I've made a first version of the recon ship (it uses the patrol boat sprite for now:

`rules/ships.yaml`
```ruby
RADARSHIP:
	Inherits: ^Ship
	Inherits@Selection: ^SelectableSupportUnit
	-AppearsOnRadar:
	Valued:
		Cost: 850
	Tooltip:
		Name: actor-radarship.name
		RequiresCondition: !deployed
	Tooltip@DEPLOYED:
		Name: actor-radarship.deployed--name
		RequiresCondition: deployed
	Buildable:
		Queue: Ship
		BuildAtProductionType: Ship
		Prerequisites: harbor, radar
		BuildPaletteOrder: 90
		Description: actor-radarship.description
	UpdatesPlayerStatistics:
		AddToArmyValue: true
	Health:
		HP: 13000
	Armor:
		Type: Steel
	Mobile:
		Speed: 75
		TurnSpeed: 40
	RevealsShroud:
		Range: 14c0
		MinRage: 7c0
		RevealGeneratedShroud: False
	RevealsShroud@Hacked:
		Range: 7c0
	RenderDetectionCircle:
		TrailCount: 8
	DetectCloaked:
		Range: 14c0
		RequiresCondition: !ecm-disabled
		DetectionTypes: Cloak, Underwater
```

`sequences/ships.yaml`  (it'd be cool if there was a radar on the boat, that turns around)
```ruby
radarship:
	idle:
		Filename: bits/patrolboat.png
		Facings: 8
	icon:
		Filename: bits/units-icons.png
		Start: 51
```

`languages/rules/en.ftl`
```yaml
actor-radarship =
   .description = Slow ship equipped with a long-range radar.n
    Can detect cloaked units.
      Unarmed
   .name = Reconnaissance Vessel
   .deployed--name = Reconnaissance Vessel (deployed)
```

I also thought that maybe a recon submarine could be made. It'd be unarmed, and underwater all the time, will still having the same range as the recon boat. Maybe it'd need the techcenter prerequisite.

**Comment by OcelotWalrus** [comment]
Some updates:

I've began working on some implementations discussed here, and I was able to do some stuff (the progress is happening [here](https://github.com/OcelotWalrus/CromhaHV):

* added a simple recon boat and submarine, and a miner ship
* added a rocket ship, who's got a turreted rocket launcher on it
* added the oil resource
* added a new conquest map, that focuses on water, and having oil reservoirs extractors

About the oil resource:

> The oil resource can only be placed on `Littoral`, `Pelagic`, `Oceanic` and `Water` terrain types, and it has no texture. It can only be recognize with random sparks, that spawns not that often (every 120 to 250 ticks, instead of 20 to 40 for normal ores). The oil resource is considered as a WaterOre terrain, and not Ore as other minerals. With that comes a new building: the Amphibious Refinery, that works exactly like the regular refinery, but can be placed on water and handles tanker vessels
>
> You can produce miner ships from the utility tab, and they will spawn at your primary harbor. You can then deploy this minership at a spotted oil reservoir, and it'll deploy as a normal miner. Starting from there, it'll work as a normal mining tower, and tanker vessels will eventually spawn and go toward your nearest Water Refinery.
>
> How I placed oil resources on the map I've made, is that it place about 6 of them close, at a random stage (one of the 3 ore levels).

If you're interesting, please clone [my fork](https://github.com/OcelotWalrus/CromhaHV) and test the naval rework changes, by playing on the new map Rain Under The Bay! I'd like to get your thoughts on that.

P.S. Sprites must be worked on + my fork also adds many other think, but that's just for me, only a few stuff developed there are for the navy rework.
